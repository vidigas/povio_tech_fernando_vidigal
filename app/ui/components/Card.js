import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

import Button from '../atoms/Button';


const renderButton = (isLiked, clickAction, isLoggedIn ) => {
  let label = isLiked ? 'unlike' : 'like' ;

 if(!isLoggedIn) return;
 else return <Button label={label} onClick={clickAction} />
}

const renderLikes = (classes, likes, clickAction, isLiked, isLoggedIn ) => {

     return (
      <div className={classes.like}>
        <span className={classes.span}> {likes} likes </span> <br/>
         {renderButton(isLiked, clickAction, isLoggedIn)}
      </div>
      );
         
}

const Card = ({key, classes, username, id, likes, clickAction, isLiked, isLoggedIn }) => {
    
  return (
    <div className={classes.container}>
      <div className={classes.user}>
        <span className={classes.username}> {username} </span> <br/>
      </div>
      {renderLikes(classes, likes, clickAction, isLiked, isLoggedIn)}
    </div>
  )
}

const style = {
  container: {
    height: '120px',
    width: '400px',
    backgroundColor:({ isLiked }) => isLiked ? '#d6f7c5': '#eaf2ff',
    display: 'flex',
    flexDirection: 'row',
    padding: '10px',
    marginBottom: '20px'
  },
  user: {
    height: '100px',
    margin: '0',
    width: '250px',
    padding: '0',
  },
  username: {
    fontSize: 'xx-large'
  },
  span: {
    marginLeft: '15px',
    fontSize: 'large'
  }
}

Card.propTypes = {
	username: propTypes.string.isRequired,
  likes: propTypes.number.isRequired,
	id: propTypes.string.isRequired,
	clickAction: propTypes.func.isRequired,
};

export default injectSheet(style)(Card);