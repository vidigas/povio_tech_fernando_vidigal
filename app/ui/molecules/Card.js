import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

import Button from '../atoms/Button';


const Card = ({ classes, username, id, likes, onClick }) => {
  return (
    <div className={classes.container}>
      <div className={classes.user}>
        <span className={classes.username}> {username} </span> <br/>
        <span className={classes.span}> {likes} likes </span> <br/>
        <span className={classes.span}> offline </span>
      </div>
       <div className={classes.like}>
      <Button label={'like'} onClick={()=> console.log('aqui vai da like')} />
      </div>

    </div>
  )
}

const style = {
  container: {
    height: '120px',
    width: '400px',
    backgroundColor: '#eaf2ff',
    display: 'flex',
    flexDirection: 'row',
    padding: '10px',
    marginBottom: '20px'
  },
  user: {
    height: '100px',
    margin: '0',
    width: '250px',
    padding: '0',
  },
  username: {
    fontSize: 'xx-large'
  },
  span: {
    marginLeft: '15px',
    fontSize: 'large'
  }
}

Card.propTypes = {
	username: propTypes.string.isRequired,
  likes: propTypes.number.isRequired,
	id: propTypes.string.isRequired,
	onClick: propTypes.func.isRequired,
};

export default injectSheet(style)(Card);