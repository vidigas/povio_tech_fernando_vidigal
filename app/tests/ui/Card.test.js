import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../ui/components/Card';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<Card username="username" likes={4} id={'abcd12343'} clickAction={()=> console.log('click')} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

