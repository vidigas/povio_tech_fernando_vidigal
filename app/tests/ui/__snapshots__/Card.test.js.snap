// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Card-container-0-1-1 Card-container-0-1-5"
>
  <div
    className="Card-user-0-1-2"
  >
    <span
      className="Card-username-0-1-3"
    >
       
      username
       
    </span>
     
    <br />
  </div>
  <div>
    <span
      className="Card-span-0-1-4"
    >
       
      4
       likes 
    </span>
     
    <br />
  </div>
</div>
`;
